core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1361,10 +1366,10 @@ class ReplicaManager(val config: KafkaConfig,
    */
   private def appendToLocalLog(internalTopicsAllowed: Boolean,
                                origin: AppendOrigin,
-                               entriesPerPartition: Map[TopicPartition, MemoryRecords],
+                               entriesPerPartition: Map[TopicIdPartition, MemoryRecords],
```

Comment:
is there a reason to pass this data structure here if we are not using the ID to check the append at the log level?

Reply:
two reasons here 
1. I didn't want to keep convert between `TopicIdPartitions` to `TopicPartition`
2. [KAFKA-16212](https://issues.apache.org/jira/browse/KAFKA-16212) will eventually use `TopicIdPartitions` to `getPartitionOrException`

Reply:
Ok -- once we start using these across the log layer it makes sense.